spec/jobs/arcgis_token_job_spec.rb

```diff
@@ -13,7 +14,8 @@
           session: {},
           sp: nil,
         ).and_return(analytics)
-      allow(ArcgisApi::Geocoder).to receive(:new).and_return(geocoder)
+      allow(ArcgisApi::GeocoderFactory).to receive(:new).and_return(geocoder_factory)
```

Comment:
bonus to switching to a class/module method: we don't need to stub `:new` (which is one of my biggest pet peeves)